src/libraries/System.Collections.Immutable/src/System.Collections.Immutable.csproj

```diff
@@ -24,16 +25,29 @@ The System.Collections.Immutable library is built-in as part of the shared frame
     <Compile Include="System\Collections\Frozen\FrozenSet.cs" />
     <Compile Include="System\Collections\Frozen\FrozenSetInternalBase.cs" />
     <Compile Include="System\Collections\Frozen\ImmutableArrayFactory.cs" />
-    <Compile Include="System\Collections\Frozen\Int32FrozenDictionary.cs" />
-    <Compile Include="System\Collections\Frozen\Int32FrozenSet.cs" />
     <Compile Include="System\Collections\Frozen\ItemsFrozenSet.cs" />
     <Compile Include="System\Collections\Frozen\KeysAndValuesFrozenDictionary.cs" />
     <Compile Include="System\Collections\Frozen\LengthBucketsFrozenDictionary.cs" />
     <Compile Include="System\Collections\Frozen\LengthBucketsFrozenSet.cs" />
-    <Compile Include="System\Collections\Frozen\OrdinalStringFrozenDictionary.cs" />
-    <Compile Include="System\Collections\Frozen\OrdinalStringFrozenSet.cs" />
+    <Compile Include="System\Collections\Frozen\String\OrdinalStringFrozenDictionary.cs" />
+    <Compile Include="System\Collections\Frozen\String\OrdinalStringFrozenSet.cs" />
+    <Compile Include="System\Collections\Frozen\SmallFrozenDictionary.cs" />
+    <Compile Include="System\Collections\Frozen\SmallFrozenSet.cs" />
     <Compile Include="System\Collections\Frozen\ValueTypeDefaultComparerFrozenDictionary.cs" />
     <Compile Include="System\Collections\Frozen\ValueTypeDefaultComparerFrozenSet.cs" />
+
+    <Compile Include="System\Collections\Frozen\Integer\IntegerFrozenDictionary.cs" Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
+    <Compile Include="System\Collections\Frozen\Integer\IntegerFrozenSet.cs" Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
+    <Compile Include="System\Collections\Frozen\Integer\SmallIntegerFrozenDictionary.cs" Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
+    <Compile Include="System\Collections\Frozen\Integer\SmallIntegerFrozenSet.cs" Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
+    <Compile Include="System\Collections\Frozen\Integer\SparseRangeIntegerFrozenSet.cs" Condition="$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
+
+    <Compile Include="System\Collections\Frozen\Int32\Int32FrozenDictionary.cs" />
+    <Compile Include="System\Collections\Frozen\Int32\Int32FrozenSet.cs" />
+    <Compile Include="System\Collections\Frozen\Int32\SmallInt32FrozenDictionary.cs" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
+    <Compile Include="System\Collections\Frozen\Int32\SmallInt32FrozenSet.cs" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
+    <Compile Include="System\Collections\Frozen\Int32\SparseRangeInt32FrozenSet.cs" Condition="!$([MSBuild]::IsTargetFrameworkCompatible('$(TargetFramework)', 'net7.0'))" />
```

Comment:
These msbuild functions aren't cost-free and we usually try to not have them per compile item. Can you please move them into a separate ItemGroup? Same for the ones above (L39-43).

Reply:
WIll do.